modules/node_modules/@ncigdc/components/Case.js

```diff
@@ -156,8 +168,16 @@ const Case = ({
     });
   }, []);
 
+  const filesToAdd = files.filter(f => !cartFiles.some(cf => cf.file_id === f.file_id));
+
   return (
     <Column spacing={theme.spacing}>
+      <Row style={{ justifyContent: 'flex-end' }}>
+        <Button onClick={() => dispatch(toggleFilesInCart(filesToAdd.length ? filesToAdd : files))}>
+          {filesToAdd.length ? 'Add all files to the cart' : 'Remove all files from the cart'}
+        </Button>
```

Comment:
filesToAdd is filtered so it is only the ones that aren't in the cart.
So if some are in the cart, it will add the rest. If none are in the cart, it will remove them all :)